### PR TITLE
First pass at Slatkin-Maddison, support for multiple compartments.

### PR DESCRIPTION
**Pull request by stephenshank**
CLI was tested with `npm link`. Data and README available. Comments/feedback welcome.

### CODE REVIEW

File: src/cli/slatkin-maddison.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  ).option(
+    '-i --iterations [iterations]',
+    'Number of bootstrap iterations (default is 10000)',
+    10000
+  ).option(
+    '-s --seed [seed]',
+    'Seed for random number generator',
+    1
+  ).option(
+    '-r --regex <regex> [otherRegexes...]',
```
Comment (stevenweaver) [inline src/cli/slatkin-maddison.js]: Regular expressions can be cumbersome to use and not entirely user friendly. A CSV would be better.

File: src/hypothesis-testing/slatkin-maddison.js [hunk 2]
```diff
@@ -1,24 +1,84 @@
...
 }
 
-/*
- * Slatkin-Maddison test
- */
-export default function slatkin_maddison(tree) {
-
-  // tree has to have an attribute
-  // 1) max parsimony, then 2) count migration events
+export default function slatkin_maddison(tree, regexes, number_of_iterations, seed) {
```
Comment (stevenweaver) [inline src/hypothesis-testing/slatkin-maddison.js]: Instead of regexes, why not just accept an object?

Comment (stephenshank) [reply src/hypothesis-testing/slatkin-maddison.js]: In hindsight, this is definitely the appropriate separation of concerns. It also made adding CSV cleaner.

**Comment by stephenshank** [comment]
@stevenweaver I've attempted to implement the requested changes. Also, please note the change to `main` in `package.json`, which I think is necessary for importing the package from node. In a clean directory,

```
npm init -y
npm install phylotree@1.0.0-alpha.10
node -e "const phylotree = require('phylotree');"
```

results in `Error: Cannot find module 'phylotree'`. The plan is to release this as `1.0.0-alpha.11` and create a phylotree bioconda package with @davebx.

**Comment by stevenweaver** [comment]
Much more work needs to be done before wrapping this and releasing to bioconda (visualization, verification, new SM-2019 implementation). A target of `March 22nd` has been firmly established, and we have a preliminary `HYPHYMP` implementation as it stands.

**Pull request merged**